### PR TITLE
add .clang-format

### PR DESCRIPTION
**Pull request by PikachuHy**
clang format config 

> we'd better follow the jolly style:
assignment: a= 1 but not a = 1
func apply: func () but not func()

https://github.com/XmacsLabs/mogan/pull/118#discussion_r841077011

### CODE REVIEW

**Comment by PikachuHy** [comment]
https://clang.llvm.org/docs/ClangFormatStyleOptions.html

**Comment by darcy-shen** [comment]
https://github.com/XmacsLabs/mogan/blob/main/src/Texmacs/Texmacs/texmacs.cpp#L91-L95
```
void 
clean_exit_on_segfault (int sig_num) {
  (void) sig_num;
  FAILED ("segmentation fault");
}
```
The return type should be put on the first line. And the second line followed by the function name.

**Comment by darcy-shen** [comment]
I suggest that you should add a script named `format` under `bin/`, in that way:

```
bin/format src/Texmacs/Texmacs/texmacs.cpp
# <=> clang-format -i src/Texmacs/Texmacs/texmacs.cpp
```
Please also commit the result of `bin/format src/Texmacs/Texmacs/texmacs.cpp`, and try to reduce unnecessary changes as much as possible.

Once it is ok, it would be better to start a discussion on the texmacs-dev mail list.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SpaceBeforeAssignmentOperators: false
+SpaceBeforeParens: Always
```
Comment (darcy-shen) [inline .clang-format:2]:
```
AlwaysBreakAfterReturnType: TopLevelDefinitions
```
Since  clang-format 3.8

Comment (darcy-shen) [reply .clang-format:2]:
```
AllowShortIfStatementsOnASingleLine: AllIfsAndElse
```
Since clang-format 9

Comment (darcy-shen) [reply .clang-format:2]:
```
AlignConsecutiveAssignments: true
```

Comment (darcy-shen) [reply .clang-format:2]:
```
SpaceAfterCStyleCast: true
```

Comment (PikachuHy) [reply .clang-format:2]: > AllowShortIfStatementsOnASingleLine: AllIfsAndElse
> AllowShortFunctionsOnASingleLine: All

annoying rule for debugging

Comment (darcy-shen) [reply .clang-format]: > annoying rule for debugging

I think so! Personally, I prefer the verbose one.

Comment (darcy-shen) [reply .clang-format]: If there is no explicit good reason, I would like to respect the history code and follow the convention.

**Review by darcy-shen — APPROVED** [review]
LGTM

**Pull request merged**